# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/D3D11YCbCrImage.cpp
# Commit: 7a990fa02246
# Full Hash: 7a990fa022462c702adacfac907a0fc178940b9d
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - P1D. Properly serialize display size when sending image over IPC. r=mattwoodrow
#   
#   The code always assumed that the size of the image with the Y plane dimensions, which, while often the case, isn't correct.
#   We remove the assertions that the display offset was always (0,0) and properly carry the actual data over IPC.
#   
# ==============================================================================

diff -r 4c7f671c7b69 -r 7a990fa02246 gfx/layers/D3D11YCbCrImage.cpp
--- a/gfx/layers/D3D11YCbCrImage.cpp	Tue Oct 20 23:29:30 2020 +0000
+++ b/gfx/layers/D3D11YCbCrImage.cpp	Tue Oct 20 23:30:04 2020 +0000
@@ -6,10 +6,10 @@
 
 #include "D3D11YCbCrImage.h"
 
+#include "YCbCrUtils.h"
 #include "gfx2DGlue.h"
-#include "YCbCrUtils.h"
+#include "mozilla/gfx/DeviceManagerDx.h"
 #include "mozilla/gfx/gfxVars.h"
-#include "mozilla/gfx/DeviceManagerDx.h"
 #include "mozilla/layers/CompositableClient.h"
 #include "mozilla/layers/CompositableForwarder.h"
 #include "mozilla/layers/TextureD3D11.h"
@@ -320,7 +320,7 @@
 DXGIYCbCrTextureAllocationHelper::DXGIYCbCrTextureAllocationHelper(
     const PlanarYCbCrData& aData, TextureFlags aTextureFlags,
     ID3D11Device* aDevice)
-    : ITextureClientAllocationHelper(gfx::SurfaceFormat::YUV, aData.mYSize,
+    : ITextureClientAllocationHelper(gfx::SurfaceFormat::YUV, aData.mPicSize,
                                      BackendSelector::Content, aTextureFlags,
                                      ALLOC_DEFAULT),
       mData(aData),
@@ -332,7 +332,7 @@
 
   DXGIYCbCrTextureData* dxgiData =
       aTextureClient->GetInternalData()->AsDXGIYCbCrTextureData();
-  if (!dxgiData || aTextureClient->GetSize() != mData.mYSize ||
+  if (!dxgiData || aTextureClient->GetSize() != mData.mPicSize ||
       dxgiData->GetYSize() != mData.mYSize ||
       dxgiData->GetCbCrSize() != mData.mCbCrSize ||
       dxgiData->GetColorDepth() != mData.mColorDepth ||
@@ -412,10 +412,10 @@
       aAllocator ? aAllocator->GetTextureForwarder() : nullptr;
 
   return TextureClient::CreateWithData(
-      DXGIYCbCrTextureData::Create(textureY, textureCb, textureCr, mData.mYSize,
-                                   mData.mYSize, mData.mCbCrSize,
-                                   mData.mColorDepth, mData.mYUVColorSpace,
-                                   mData.mColorRange),
+      DXGIYCbCrTextureData::Create(textureY, textureCb, textureCr,
+                                   mData.mPicSize, mData.mYSize,
+                                   mData.mCbCrSize, mData.mColorDepth,
+                                   mData.mYUVColorSpace, mData.mColorRange),
       mTextureFlags, forwarder);
 }
 